src/controllers/notifications/NotificationController.cpp

```diff
@@ -98,20 +98,23 @@ void NotificationController::removeChannelNotification(
 void NotificationController::playSound()
 {
     static auto player = new QMediaPlayer;
-    static QUrl currentPlayerUrl;
 
     QUrl highlightSoundUrl =
         getSettings()->notificationCustomSound
             ? QUrl::fromLocalFile(
                   getSettings()->notificationPathSound.getValue())
             : QUrl("qrc:/sounds/ping2.wav");
 
-    if (currentPlayerUrl != highlightSoundUrl)
+    // set media if not loading, loaded, stalled, buffering, or fully buffered
+    if (!(player->mediaStatus() == QMediaPlayer::LoadingMedia ||
+          player->mediaStatus() == QMediaPlayer::LoadedMedia ||
+          player->mediaStatus() == QMediaPlayer::StalledMedia ||
+          player->mediaStatus() == QMediaPlayer::BufferingMedia ||
+          player->mediaStatus() == QMediaPlayer::BufferedMedia))
```

Comment:
mediaStatus() will be either `QMediaPlayer::NoMedia` after QMediaPlayer was just created or `QMediaPlayer::EndOfMedia` at each successive call when ping sound finished [docs](https://doc.qt.io/qt-6.2/qmediaplayer.html#MediaStatus-enum), so both if's can be simplified.

also, setMedia() will be called at almost every ping now instead of only when audio file changed, would be cool to avoid that.

Reply:
You're right, I did some testing and found that when there's a ping, `mediaStatus()` will be `QMediaPlayer::NoMedia`, then when media is set it will be `QMediaPlayer::LoadingMedia`.

After a successful ping, `mediaStatus()` will be as you wrote `QMediaPlayer::EndOfMedia`.

After a few pings `mediaStatus()` becomes `QMediaPlayer::BufferedMedia` just right before `play()` is called and if media is not set, the ping sound will stop playing until media is set.

So the if statement should instead be:

```
    // set media if no media, or if media is buffered
    if (player->mediaStatus() == QMediaPlayer::NoMedia ||
        player->mediaStatus() == QMediaPlayer::BufferedMedia)
```

Right?

That should considerably reduce the amount of times `setMedia()` is called.

However, what I don't understand is why would the ping sound not play at all when `mediaStatus()` is to my understanding already buffered and ready to play again?
Or am I misunderstanding the role of `QMediaPlayer::BufferedMedia`?

Reply:
tbh, hard for me to suggest good solution here because i dont know root cause of this bug and can't reproduce it on windows.
If your approach fixed ping sounds stopping, it would suggest calling `setMedia()` more often than needed is the key here, but its more of a workaround than addressing root cause. 
It could be many things, including qt bug or system problem, so would be best if more Arch users confirm issue and take a look into this.

Reply:
I'm fine if this solution is not a fix, but a stopgap solution. Qt6 has changed the multimedia API so we'll need to revisit this shortly anyway.
If this change doesn't break anything, only adding some more processing requirements on other platforms I'm ok with that tradeoff if it means pings will be more consistent